### PR TITLE
DomainParticipant create_publisher

### PR DESCRIPTION
**Pull request by laura-eprosima**
* Add new create_publisher method following DDS standard
* New unit test to check PIM create_publisher functionality
* Implementation of PSM Publisher constructor
* New unit test for PSM constructor

NOTE: This PR must be merged after #1073

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1633/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1399/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1629/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1647/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1413/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1643/)

File: src/cpp/fastdds/domain/DomainParticipantImpl.cpp [hunk 1]
```diff
@@ -213,38 +213,6 @@ Publisher* DomainParticipantImpl::create_publisher(
...
-        return nullptr;
-    }
-
-    if (!qos.publisher_attr.remoteLocatorList.isValid())
-    {
-        logError(PARTICIPANT, "Remote Locator List for Publisher contains invalid Locator");
-        return nullptr;
-    }
-
-    if (!qos.check_qos() || !qos.publisher_attr.qos.checkQos() || !qos.publisher_attr.topic.checkQos())
```
[the remark this answers is not in the document]
Comment (laura-eprosima) [reply src/cpp/fastdds/domain/DomainParticipantImpl.cpp]: It is not necessary as the function only returns true, and it is going to disappear once the set/get_qos for the publisher is implemented

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1796/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1547/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1790/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1812/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1560/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1806/)

**Comment by richiware** [comment]
Failed tests not related. Known flaky tests.

**Comment by LuisGP** [comment]
This PR broke `DDSSimpleCommunicationTypeDiscovery` test. Isn't a flaky test anymore, is totally broken.
The publisher needs the QoS information from `publisher_attributes` (as `PublisherQos` I guess).
I'm not sure, but I think that the topic information is only needed by the datawriter.

**Pull request merged**